client/src/components/pages/Profile.js

```diff
@@ -0,0 +1,185 @@
+    setUpdatedDetails({ ...updatedDetails, [field]: e.target.value })
+  }
+
+  const updateUserProfile = (e) => {
+    e.preventDefault()
+    const body = {
+      id: loggedInUser.id,
+      bio: updatedBio,
+      occupation: updatedOccupation,
+    }
+    updateProfile(body)
+    toggleEditing(e, false)
+  }
+
+  const returnDetails = () => {
+    return editing ? (
+      <div
+        className='user-details border border-light rounded bg-light py-4 px-3 edit-container'
+        id='user-details'
+      >
+        <h6>Age</h6>
+        <p className='ml-2'>{age}</p>
+        <h6>Birthday</h6>
+        <p className='ml-2'>
+          {month} {day}, {year}
+        </p>
+        <form onSubmit={updateUserProfile} id='update-profile-form form-group'>
+        <div className='d-flex justify-content-between'>
+            <h6>Occupation</h6>
+            <h6 className='text-muted'>{updatedOccupation.length}/100</h6>
+          </div>
+          <input
+            onChange={(e) => updateInput(e, 'updatedOccupation')}
+            type='text'
+            name='occupation'
+            id='occupation-edit-field'
+            className='form-control mb-3'
+            placeholder='occupation'
+            value={updatedOccupation ? updatedOccupation : ''}
+            maxLength='150'
+          />
+          <div className='d-flex justify-content-between'>
+            <h6>Bio</h6>
+            <h6 className='text-muted'>{updatedBio.length}/500</h6>
+          </div>
+          <textarea
+            onChange={(e) => updateInput(e, 'updatedBio')}
+            type='text'
+            name='bio'
+            id='bio-edit-field'
+            className='bio-text mb-3'
+            placeholder='bio'
+            value={updatedBio ? updatedBio : ''}
+            maxLength='500'
+          />
+          <div
+            className='form-group row my-3'
+          >
+            <div className='col'>
+              <button
+                className='btn btn-primary btn-block'
+                onClick={(e) => toggleEditing(e, false)}
+              >
+                Cancel
+              </button>
+            </div>
+            <div className='col'>
+              <button className='btn btn-primary btn-block' id='submit'>
+                Save
+              </button>
+            </div>
+          </div>
+        </form>
+      </div>
+    ) : (
+      <div>
+        <div
+          className='user-details border border-light rounded bg-light py-4 px-3'
+          id='user-details'
+        >
+          <h6>Age</h6>
+          <p className='ml-2'>{age}</p>
+          <h6>Birthday</h6>
+          <p className='ml-2'>
+            {month} {day}, {year}
+          </p>
+          <h6>Occupation</h6>
+          {occupation ? <p className='ml-2'>{occupation}</p> : <br></br>}
+          <h6>Bio</h6>
+          {bio ? <p className='ml-2'>{bio}</p> : <br></br>}
+          {loggedInUser.id === id ? (
+            <button
```

Comment:
this button is missing an ID, I would make sure that all your interactable elements have IDs with them, this will lead to easier testing in the feature for the full suite of tests (unit, integration, e2e) if you ever get that far